job.go

```diff
@@ -595,11 +603,40 @@ func WithStartDateTime(start time.Time) StartAtOption {
 		if start.IsZero() || start.Before(time.Now()) {
 			return ErrWithStartDateTimePast
 		}
+		if !j.stopTime.IsZero() && j.stopTime.Before(start) {
+			return ErrStartTimeLaterThanEndTime
+		}
 		j.startTime = start
 		return nil
 	}
 }
 
+// WithStopAt sets the option for stopping the job at
+// a specific datetime.
+func WithStopAt(option StopAtOption) JobOption {
+	return func(j *internalJob) error {
+		return option(j)
+	}
+}
+
+// StopAtOption defines options for stopping the job
+type StopAtOption func(*internalJob) error
```

Comment:
Once my PR is merged, could you pull it in and change the signature here to accept a time.Time as well, and pass that in to avoid using time.Now() for validation inside the WithStopDateTime func?